scheduler/src/cook/kubernetes/compute_cluster.clj

```diff
@@ -247,15 +241,26 @@
   cc/ComputeCluster
   (launch-tasks [this offers task-metadata-seq]
     (doseq [task-metadata task-metadata-seq]
-      (let [pod-namespace (get-namespace-from-task-metadata namespace-config task-metadata)]
+      ; Has the workload of launching tasks changed or has the cost of doing them changed?
+      ; Note we can't use timer/time! because it wraps the body in a Callable, which rebinds 'this' to another 'this'
+      ; causing breakage.
+      ; In addition, it captures controller/update-cook-expected-state and prevents with-redefs on
+      ; update-cook-expected-state from working correctly in unit tests.
+      (let [timer-context (timers/start (metrics/timer "compute-cluster-launch-tasks" name))
+            pod-namespace (get-namespace-from-task-metadata namespace-config task-metadata)]
         (controller/update-cook-expected-state
           this
           (:task-id task-metadata)
           {:cook-expected-state :cook-expected-state/starting
-           :launch-pod {:pod (api/task-metadata->pod pod-namespace name task-metadata)}}))))
+           :launch-pod {:pod (api/task-metadata->pod pod-namespace name task-metadata)}})
+        (.stop timer-context))))
 
   (kill-task [this task-id]
-    (controller/update-cook-expected-state this task-id {:cook-expected-state :cook-expected-state/killed}))
+    ; Note we can't use timer/time! because it wraps the body in a Callable, which rebinds 'this' to another 'this'
+    ; causing breakage.
+    (let [timer-context (timers/start (metrics/timer "compute-cluster-kill-tasks" name))]
+      (controller/update-cook-expected-state this task-id {:cook-expected-state :cook-expected-state/killed})
+      (.stop timer-context)))
```

Comment:
Should we put this in a finally to make sure exceptions don't blow past it?

Reply:
Already done. Its wrapped in safe-kill-task (which also does the locking to prevent the kill-launch race), which all callers go through.

Reply:
If an exception is thrown in `controller/update-cook-expected-state`, won't it cause the timer to not stop?

Reply:
Just means we won't log a metric, which is intentional.